File: swift/easy/1474-delete-n-nodes-after-m-nodes-of-a-linked-list.swift
func deleteNodes(_ head: ListNode?, _ m: Int, _ n: Int) -> ListNode? {
    var dummy = ListNode(0)
    dummy.next = head

    var current: ListNode? = dummy
    while (current != nil) {
        var i = 0
        while current != nil && i < m {
            current = current?.next
            i += 1
        }

        i = 0
        while current != nil && i < n {
            current!.next = current!.next?.next
            i += 1
        }
    }

    return dummy.next
}

File: swift/easy/1474.delete-n-nodes-after-m-nodes-of-a-linked-list.java
public ListNode deleteNodes(ListNode head, int m, int n) {
    ListNode dummy = new ListNode(0, head);

    ListNode current = dummy;
    while (current.next != null) {
        int i = 0;
        while (current.next != null && i != m) {
            current = current.next;
            i++;
        }

        i = 0;
        while (current.next != null && i != n) {
            current.next = current.next.next;
            i++;
        }
    }

    return dummy.next;
}
